app/components/SharingButtons.tsx

```diff
@@ -0,0 +1,91 @@
+import { ReactNode } from 'react';
+import { FaTwitter, FaFacebook, FaEnvelope } from 'react-icons/fa';
+
+enum Network {
+  Facebook,
+  Twitter,
+  Email
+}
+
+function linkName(network: Network): string {
+  switch (network) {
+  case Network.Facebook:
+    return 'Facebook';
+  case Network.Twitter:
+    return 'Twitter';
+  case Network.Email:
+    return 'Email';
+  }
+}
+
+function colors(network: Network): [string, string] {
+  switch (network) {
+  case Network.Facebook:
+    return ['#3b5998', '#2d4373'];
+  case Network.Twitter:
+    return ['#55acee', '#2795e9'];
+  case Network.Email:
+    return ['#777777', '#5e5e5e'];
+  }
+}
+
+function url(network: Network, path: string, text: string): string {
+  const urlToShare = encodeURIComponent('https://crosshare.org' + path);
+  const textToShare = encodeURIComponent(text);
+  switch (network) {
+  case Network.Facebook:
+    return 'https://facebook.com/sharer/sharer.php?u=' + urlToShare;
+  case Network.Twitter:
+    return 'https://twitter.com/intent/tweet/?text=' + textToShare + '&url=' + urlToShare;
+  case Network.Email:
+    return 'mailto:?subject=' + textToShare + '&body=' + urlToShare;
+  }
+}
+
+function icon(network: Network): ReactNode {
+  switch (network) {
+  case Network.Facebook:
+    return <FaFacebook css={{ verticalAlign: 'text-bottom', }} />;
+  case Network.Twitter:
+    return <FaTwitter css={{ verticalAlign: 'text-bottom', }} />;
+  case Network.Email:
+    return <FaEnvelope css={{ verticalAlign: 'text-bottom', }} />;
+  }
+}
+
+interface SharingButtonProps extends SharingButtonsProps {
+  network: Network
+}
+
+function SharingButton({ network, path, text }: SharingButtonProps) {
+  return <a css={{
+    display: 'inline-block',
+    margin: '0 0.2em',
+    padding: '0.2em 0.5em',
+    borderRadius: '0.3em',
+    color: 'var(--social-text)',
+    //textShadow: '1px 1px 1px #333',
+    backgroundColor: colors(network)[0],
```

Comment:
In dark mode, I'm seeing that the text for Twitter and Email buttons have 1.81 and 3.29 contrast.
<img width="308" alt="image" src="https://user-images.githubusercontent.com/7526790/226455687-17040b6e-935e-4ce2-9116-50f5f3e379e8.png">

Reply:
Good catch. I'm going to either find a way to fix the colors or do some design tweaks to the buttons to get that readable